### PR TITLE
ORC-827: Utilize Array copyOf

### PR DESCRIPTION
**Pull request by belugabehr**
### What changes were proposed in this pull request?
Utilize JDK Array copyOf

### Why are the changes needed?
Less code, small performance boost perhaps

### How was this patch tested?
Existing unit tests as functionality has not changed.

### CODE REVIEW

File: java/core/src/java/org/apache/orc/InMemoryKeystore.java [hunk 1]
```diff
@@ -278,16 +276,7 @@ public InMemoryKeystore addKey(String keyName, int version,
...
-    final byte[] buffer = new byte[algorithm.keyLength()];
-    if (algorithm.keyLength() > masterKey.length) {
-
-      System.arraycopy(masterKey, 0, buffer, 0, masterKey.length);
-      /* fill with zeros */
-      Arrays.fill(buffer, masterKey.length, buffer.length - 1, (byte) 0);
-
-    } else {
-      System.arraycopy(masterKey, 0, buffer, 0, algorithm.keyLength());
-    }
```
Comment (belugabehr) [inline java/core/src/java/org/apache/orc/InMemoryKeystore.java:290]: As I understand this code, it's a bit superfluous.  There is no need to "fill with zeros" as the array will, by default in Java, pre-populated with zero value.

And this:

> For any indices that are valid in the copy but not the original, the copy will contain (byte)0. Such indices will exist if and only if the specified length is greater than that of the original array.

https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#copyOf(byte[],%20int)

**Comment by dongjoon-hyun** [comment]
cc @omalley

**Review by dongjoon-hyun — COMMENTED** [review]
BTW, @belugabehr . Although you don't have your patch in Apache Orc 1.6.9, could you participate Apache ORC 1.6.9 RC0 vote?
- https://lists.apache.org/thread.html/rab78879c2ff44b0357d49b592974545d5e649852c53a696510e19bdd%40%3Cdev.orc.apache.org%3E

**Comment by belugabehr** [comment]
@dongjoon-hyun Are you able to assist in moving this in?

Thanks!

**Comment by belugabehr** [comment]
@dongjoon-hyun This one has been getting a bit stale.  What additional details can I provide?

Thanks!!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs)

**Comment by dongjoon-hyun** [comment]
Merged to main for Apache ORC 1.9.0.

**Pull request closed without merging**